### PR TITLE
Dockerizing Ledger server (rebased PR)

### PR DESCRIPTION
**Pull request by willy-b**
Docker makes life easier, so let's use it.

This 1st pass Dockerization should let you set up a Ledger server on any machine running the Docker daemon without worrying about installing correct versions of NodeJS/mongo/redis/etc.

Steps to run Ledger server:
0. _One time:_ Setup config variables.
  Edit `config/config.$NODE_ENV.js` to set your BitGo / GitHub credentials. (The BitGo and GitHub environment variable overrides will still work if you want them.) Set `export EXTERNAL_PORT=80` or whatever you want.
1. Run `docker-compose build` (wait a bit; later we could pull this image from a docker-repo)
2. Run `./run_docker.sh`

The Ledger server should be running on port $EXTERNAL_PORT (set in step 0).

Only dependencies should be `docker` and `docker-compose`.

You can get install info for those here:
  Installing docker: https://docs.docker.com/engine/installation/
  Installing docker-compose: https://docs.docker.com/compose/install/

Caveats:
- data is NOT PERSISTENT by default
- need to setup HTTPS by default

This replaces the previous PR #8

### CODE REVIEW

**Comment by mrose17** [comment]
@willy-b - just checking? are any more commits coming for a while, or can i start the next review? thanks!

**Comment by willy-b** [comment]
Hey @mrose17, this is it! Go ahead and review when you have time :-)

File: gulpfile.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ gulp.task('run', function () {
   nodemon({
     script: 'src/index.js',
     ext: 'js',
-    env: { 'DEBUG': '*' },
+    /** env: { 'DEBUG': '*' }, **/
```
Comment (ayumi) [inline gulpfile.js:28]: what happens when this is un/commented? maybe a comment explaining that would be useful

Comment (willy-b) [reply gulpfile.js:28]: I agree. That is confusing -- I'll add a comment along with fixes for whatever other review items are added here (by you or @mrose17).
The explanation is in the last PR for now: https://github.com/brave/ledger/pull/8#discussion_r81400872

**Review by mrose17 — APPROVED** [review]
thanks for the reworking. i am merging now!

**Pull request merged**